configs/components/marionette-collective.rb

```diff
@@ -66,7 +66,33 @@
   when "aix"
     pkg.install_service "resources/aix/mcollective.service", nil, "mcollective"
   when "windows"
-    puts "Service files not enabled on windows"
+    service_hash = [{
+      :directory_ref => settings[:bindir_id],
+      :id => "MCOService",
+      :guid => "7601FCEA-90B3-CC69-6A69-4087FBC7292D",
+      :win64 => settings[:win64],
+      :file => {
+        :id => "RubyWExe",
+        :source => settings[:rubyw],
+      },
+      # This service is installed with start set to demand because
+      # it won't be correctly configured to start right away. The
+      # puppet run that configures mcollective will allow it to start
+      # and set it to automatic
+      :serviceinstall => {
+        :id => "MCOServiceInstaller",
+        :description => "Puppet Labs server orchestration framework",
+        :displayname => "Marionette Collective Server",
+        :name => "mcollective",
+        :start => "demand",
+        :arguments => '-I&quot;[INSTALLDIR]\\lib&quot; -rubygems &quot;[INSTALLDIR]\\bin\\mcollectived&quot; --daemonize',
```

Comment:
This is a the beginning of the actual directory hierarchy for the current puppet agent MSI:

```
directory                            Parent directory     Target Directory
--------------------------------------------------------------------------------------------------------------------
CommonAppDataFolder                  TARGETDIR             CommonAppDataFolder
ProgramFiles64Folder                 TARGETDIR             ProgramFiles64Folder
ProgramMenuFolder                    TARGETDIR             ProgramMenuFolder
TARGETDIR                            TARGETDIR
PuppetLabsAppData                    CommonAppDataFolder   CommonAppDataFolder\PuppetLabs
CodeDir                              PuppetLabsAppData     CommonAppDataFolder\PuppetLabs\code
FacterAppData                        PuppetLabsAppData     CommonAppDataFolder\PuppetLabs\facter
MCOAppData                           PuppetLabsAppData     CommonAppDataFolder\PuppetLabs\mcollective
PuppetAppData                        PuppetLabsAppData     CommonAppDataFolder\PuppetLabs\puppet
EnvironmentsDir                      CodeDir               CommonAppDataFolder\PuppetLabs\code\environments
HieraDataDir                         CodeDir               CommonAppDataFolder\PuppetLabs\code\hieradata
CodeModulesDir                       CodeDir               CommonAppDataFolder\PuppetLabs\code\modules
ProductionDir                        EnvironmentsDir       CommonAppDataFolder\PuppetLabs\code\environments\production
ManifestsDir                         ProductionDir         CommonAppDataFolder\PuppetLabs\code\environments\production\manifests
ModulesDir                           ProductionDir         CommonAppDataFolder\PuppetLabs\code\environments\production\modules
FactsDotD                            FacterAppData         CommonAppDataFolder\PuppetLabs\facter\facts.d
MCOConfDir                           MCOAppData            CommonAppDataFolder\PuppetLabs\mcollective\etc
MCOVarDir                            MCOAppData            CommonAppDataFolder\PuppetLabs\mcollective\var
MCOLogDir                            MCOVarDir             CommonAppDataFolder\PuppetLabs\mcollective\var\log
PuppetConfDir                        PuppetAppData         CommonAppDataFolder\PuppetLabs\puppet\etc
PuppetVarDir                         PuppetAppData         CommonAppDataFolder\PuppetLabs\puppet\var
PuppetLabs                           ProgramFiles64Folder  ProgramFiles64Folder\Puppet Labs
INSTALLDIR                           PuppetLabs            ProgramFiles64Folder\Puppet Labs\Puppet
bin                                  INSTALLDIR            ProgramFiles64Folder\Puppet Labs\Puppet\bin
dirC24A635189E52DF456E2FBC3F74A024A  INSTALLDIR            ProgramFiles64Folder\Puppet Labs\Puppet\bin
dir59A60B7031D17F6FAE34CBA46C04A779  INSTALLDIR            ProgramFiles64Folder\Puppet Labs\Puppet\facter

```